pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -123,25 +127,34 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		infinispan.Status.Conditions = []infinispanv1.InfinispanCondition{}
 	}
 
+	// Check x-site configuration first.
+	// Must be done before creating any Infinispan resources,
+	// because remote site host:port combinations need to be injected into Infinispan.
+	reqLogger.Info("check if cross-site is configured")
+	xsite, err := r.computeXSite(infinispan, reqLogger)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	//xsite := &ispnutil.XSite{}
```

Comment:
@galderz  I can merge this PR. Do you want to clean up this comment and squash the commits?

Reply:
Yeah, let me cleanup and squash things first.